apps/mochi-web/components/Profile/stores/useProfileTransactionStore.ts

```diff
@@ -0,0 +1,197 @@
+import { create } from 'zustand'
+import { API, GET_PATHS } from '~constants/api'
+import type { Tx } from '~cpn/TransactionTable'
+import { transform } from '~cpn/TransactionTable/utils'
+import { MOCHI_PAY_WSS } from '~envs'
+
+export const DEFAULT_PAGE_SIZE = 15
+
+type Filters = {
+  platform?: string
+  chainId?: string
+  actions?: string[]
+}
+
+interface State {
+  profileId?: string
+  txns: Tx[][]
+  addNewTx: (tx: Tx) => void
+  loading: boolean
+  fetchTxns: (
+    filters?: Filters,
+    sort?: string,
+    page?: number,
+    size?: number,
+  ) => Promise<void>
+
+  size: number
+  setSize: (size: number) => void
+  page: number
+  setPage: (page: number) => void
+  total?: number
+
+  filters: Filters
+  setFilters: (partialFilters: Partial<Filters>) => void
+
+  sort: string
+  setSort: (sort: string) => void
+
+  initWs: (id?: string, override?: boolean) => void
+  ws: WebSocket | null
+}
+
+export const useProfileTransactionStore = create<State>((set, get) => ({
```

Comment:
Why don't we custom `useTransactionStore` but clone a new one for the profile page?